care/emr/api/viewsets/medication_request.py

```diff
@@ -8,15 +13,26 @@
     InternalQuestionnaireRegistry,
 )
 from care.emr.resources.medication.request.spec import (
+    MedicationRequestDiscontinueRequest,
     MedicationRequestReadSpec,
     MedicationRequestSpec,
+    MedicationRequestStatus,
 )
 from care.emr.resources.questionnaire.spec import SubjectType
 from care.security.authorization import AuthorizationController
 
 
 class MedicationRequestFilter(filters.FilterSet):
     encounter = filters.UUIDFilter(field_name="encounter__external_id")
+    status = filters.MultipleChoiceFilter(
+        field_name="status", choices=MedicationRequestStatus.choices()
+    )
+    is_prn = filters.BooleanFilter(method="filter_as_needed_boolean")
+
+    def filter_as_needed_boolean(self, queryset, name, value):
```

Comment:
We will not query JSON, rebuild the model or remove the filter.

Reply:
Created a new field in the model to record as_needed_boolean, which was initially captured in dosage instructions, a json field